pkg/credentialprovider/azure/azure_acr_helper.go

```diff
@@ -178,10 +180,15 @@ func performTokenExchange(
 	}
 
 	var content []byte
-	if content, err = ioutil.ReadAll(exchange.Body); err != nil {
+	limitedReader := &io.LimitedReader{R: exchange.Body, N: maxReadLength}
+	if content, err = ioutil.ReadAll(limitedReader); err != nil {
 		return "", fmt.Errorf("Www-Authenticate: error reading response from %s", authEndpoint)
 	}
 
+	if limitedReader.N <= 0 {
+		return "", errors.New("the read limit is reached")
```

Comment:
If we're confident that 10MB is sufficient for the foreseeable future, we'd better panic here.

Reply:
a library function like this should return an error. callers can panic if they wish, but error propagation is more orderly

Reply:
Should I send out a PR in the caller ?
failure to perform token exchange is a fatal error.

Reply:
I don't see anything fundamentally different about this error. I'd expect the existing error handling to be sufficient

Reply:
Here is how the func is called:
```
			cred, err := getACRDockerEntryFromARMToken(a, loginServer)
			if err != nil {
				continue
```
I don't see how the above would lead to panic.

Reply:
I don't think a panic is appropriate there. I would think the error should be propagated

Reply:
I think returning an error is more appropriate here.
FYI, I am working on adding a new util function to cover this feature in https://github.com/kubernetes/kubernetes/pull/76518/files to avoid code duplication. Currently there are about a dozen usage of `ioutil.ReadAll` without a read limit. 

```
// ReadAllWithLimit reads up to `limit` bytes from `r`, and reports an error when
// `limit` bytes are read.
func ReadAllWithLimit(r io.Reader, limit int64) ([]byte, error) {
	limitedReader := &io.LimitedReader{R: r, N: limit}
	data, err := ioutil.ReadAll(limitedReader)
	if limitedReader.N <= 0 {
		return nil, errors.New("the read limit is reached")
	}
	return data, err
}
```